[BUG]: TestAccGithubMembership fails on GHES

### Expected Behavior

Running the unit test `make testacc T=TestAccGithubTeamMembership` for github enterprise server should be successful.

### Actual Behavior

`make testacc T=TestAccGithubTeamMembership` fails because it the api request is being sent to the web frontend instead of the api, which results in a redirect to the login page and a http/406 error.

The endpoint of our enterprise server is https://_hostname_/api/v3
 
(github hostname and username has been redacted), but from the test result we can see the wrong uri is being called.
```bash
❯ make testacc T=TestAccGithubTeamMembership

==> Running acceptance tests on branch: 🌿 bug/base-url-ghes 🌿...
TF_ACC=1 CGO_ENABLED=0 go test ./github/... -v -run '^TestAcc' -run='TestAccGithubTeamMembership'  -timeout 120m -count=1
=== RUN   TestAccGithubTeamMembership
=== RUN   TestAccGithubTeamMembership/creates_a_team_membership
    resource_github_team_membership_test.go:26: Step 1/2 error: Check failed: Check 1/4 error: GET https://_hostname_/login?return_to=https%3A%2F%2F_hostname_%2Forganizations%2F0%2Fteam%2F37%2Fmemberships%2F_username_: 406  []
    panic.go:694: Error running post-test destroy, there may be dangling resources: GET https://_hostname_/login?return_to=https%3A%2F%2F_hostname_%2Forganizations%2F0%2Fteam%2F37%2Fmemberships%2F_username_: 406  []
=== RUN   TestAccGithubTeamMembership/is_case_insensitive
    resource_github_team_membership_test.go:64: Step 1/2 error: Check failed: Check 1/1 error: GET https://_hostname_/login?return_to=https%3A%2F%2F_hostname_%2Forganizations%2F0%2Fteam%2F39%2Fmemberships%2F_username_: 406  []
    panic.go:694: Error running post-test destroy, there may be dangling resources: GET https://_hostname_/login?return_to=https%3A%2F%2F_hostname_%2Forganizations%2F0%2Fteam%2F39%2Fmemberships%2F_username_: 406  []
--- FAIL: TestAccGithubTeamMembership (23.31s)
    --- FAIL: TestAccGithubTeamMembership/creates_a_team_membership (11.85s)
    --- FAIL: TestAccGithubTeamMembership/is_case_insensitive (11.46s)
FAIL
FAIL    github.com/integrations/terraform-provider-github/v6/github     23.315s
FAIL
make: *** [GNUmakefile:62: testacc] Error 1
```

### Terraform Version

N.A.

### Affected Resource(s)

Testcases which call `getTestMeta()`

### Terraform Configuration Files

```hcl

```

### Steps to Reproduce

Running below testcase against a GHES environment.
```
make testacc T=TestAccGithubTeamMembership
```

### Debug Output

```shell

```

### Panic Output

```shell

```

### Code of Conduct

- [x] I agree to follow this project's Code of Conduct

--- a/github/acc_test.go
+++ b/github/acc_test.go
@@ -36,6 +36,7 @@ type testAccConfig struct {
 	// Target configuration
 	legacyClient bool
 	baseURL      *url.URL
+	isGHES       bool
 
 	// Auth configuration
 	authMode          testMode
@@ -109,7 +110,7 @@ func TestMain(m *testing.M) {
 		u = DotComAPIURL
 	}
 
-	baseURL, _, err := getBaseURL(u)
+	baseURL, isGHES, err := getBaseURL(u)
 	if err != nil {
 		fmt.Printf("Error parsing base URL: %s\n", err)
 		os.Exit(1)
@@ -118,6 +119,7 @@ func TestMain(m *testing.M) {
 	config := testAccConfig{
 		legacyClient:                      os.Getenv("GITHUB_LEGACY_CLIENT") != "false",
 		baseURL:                           baseURL,
+		isGHES:                            isGHES,
 		authMode:                          authMode,
 		testPublicRepository:              "terraform-provider-github",
 		testPublicRepositoryOwner:         "integrations",
@@ -210,7 +212,12 @@ func getTestAppToken() (string, error) {
 		return "", fmt.Errorf("app auth not configured")
 	}
 
-	appToken, err := GenerateOAuthTokenFromApp(testAccConf.baseURL, testAccConf.appID, testAccConf.appInstallationID, testAccConf.appPEM)
+	restAPIPath := ""
+	if testAccConf.isGHES {
+		restAPIPath = GHESRESTAPIPath
+	}
+
+	appToken, err := GenerateOAuthTokenFromApp(testAccConf.baseURL.JoinPath(restAPIPath), testAccConf.appID, testAccConf.appInstallationID, testAccConf.appPEM)
 	if err != nil {
 		return "", err
 	}
@@ -228,10 +235,16 @@ func getTestToken() (string, error) {
 
 func getTestMeta() (*Owner, error) {
 	config := &Config{
+		RESTAPIPath:    "",
 		GraphQLAPIPath: "graphql",
 		LegacyClient:   testAccConf.legacyClient,
 		BaseURL:        testAccConf.baseURL,
 		Owner:          testAccConf.owner,
+	}
+
+	if testAccConf.isGHES {
+		config.RESTAPIPath = GHESRESTAPIPath
+		config.GraphQLAPIPath = GHESGraphQLAPIPath
 	}
 
 	if config.LegacyClient || testAccConf.appID == "" {

--- a/github/acc_test.go
+++ b/github/acc_test.go
@@ -109,7 +109,7 @@ func TestMain(m *testing.M) {
 		u = DotComAPIURL
 	}
 
-	baseURL, err := url.Parse(u)
+	baseURL, _, err := getBaseURL(u)
 	if err != nil {
 		fmt.Printf("Error parsing base URL: %s\n", err)
 		os.Exit(1)

